src/app/static/src/js/components/runReport/runReport.vue

```diff
@@ -162,6 +164,19 @@
                         }
                     })
                 }
+                this.validateParams()
+            },
+            validateParams() {
```

Comment:
It feels like validating would be cleaner with `isValidParam` as a computed property based on paramerValues rather a data value set in a method that you have to remember to call. Actually disabling the button could then use `!disableRun && isValidParam`. Similarly `paramError` could be computed based on `paramsAreValid` if it's always the same string.

This would mean that you would see validation errors immediately if no defaults - but I think that's ok.

Reply:
Would it make more sense for the `parameter-list` component to validate the parameters itself? At the moment it passes parameters to parent, which validates them, and then passes the error message back into the component - which isn't a very clear separation of concerns.

Either way, I agree with run button being immediately disabled if the report has parameters with no default values. This would also reduce the number of test cases.

One potential usability issue with the current implementation is that if you type a value in a required parameter field the run button doesn't become enabled until that field loses focus.